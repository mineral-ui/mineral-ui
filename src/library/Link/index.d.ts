declare module '@targetx/mineral-ui/Link';
