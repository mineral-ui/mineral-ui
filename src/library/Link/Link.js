/* @flow */
import React from 'react';
import { createStyledComponent } from '../styles';

type Props = {
  /** Content of the Link */
  children?: React$Node,
  /** A URL or URL fragment to which the Link points */
  href?: string,
  /** Element to be used as the root node - e.g. `a` or `ReactRouterLink` */
  element?: $FlowFixMe, // Should allow string | React class
  /** Available variants */
  variant?: 'regular' | 'danger' | 'success' | 'warning'
};

export const componentTheme = (baseTheme: Object) => ({
  Link_borderColor_focus: baseTheme.borderColor_theme_focus,
  Link_color: baseTheme.color_theme,
  Link_color_active: baseTheme.color_theme_active,
  Link_color_hover: baseTheme.color_theme_hover,
  Link_color_focus: baseTheme.color_theme_focus,

  ...baseTheme
});

const linkStyles = ({ variant, theme: baseTheme }) => {
  let theme = componentTheme(baseTheme);

  if (variant !== 'regular') {
    // prettier-ignore
    theme = {
      ...theme,
      Link_borderColor_focus: baseTheme[`borderColor_${variant}_focus`],
      Link_color: baseTheme[`color_${variant}`],
      Link_color_hover: baseTheme[`color_${variant}_hover`],
      Link_color_focus: baseTheme[`color_${variant}_focus`],
      Link_color_active: baseTheme[`color_${variant}_active`],
    };
  }

  return {
    color: theme.Link_color,
    textDecoration: 'none',

    '&:hover': {
      color: theme.Link_color_hover,
      textDecoration: 'underline'
    },
    '&:focus': {
      color: theme.Link_color_focus,
      outline: `1px solid ${theme.Link_borderColor_focus}`,
      outlineOffset: '2px'
    },
    // `:active` must be last, to follow LVHFA order:
    // https://developer.mozilla.org/en-US/docs/Web/CSS/:active
    '&:active': {
      color: theme.Link_color_active
    }
  };
};

/**
 * The Link component creates a hyperlink to external pages, files, anchors on the same page, or another URL.
 */
export default function Link({
  children,
  element = 'a',
  variant = 'regular',
  ...restProps
}: Props) {
  const rootProps = {
    variant,
    ...restProps
  };
  const Root = createStyledComponent(element, linkStyles, {
    displayName: 'Link',
    filterProps: ['variant']
  });

  return <Root {...rootProps}>{children}</Root>;
}
