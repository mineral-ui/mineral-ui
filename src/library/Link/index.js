/* @flow */
export { default } from './Link';
