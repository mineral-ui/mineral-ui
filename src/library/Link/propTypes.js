/* @flow */
import { bool, node, oneOf, string } from 'prop-types';
import { VARIANT } from './constants';

export const linkPropTypes = {
  /* TargetX Custom Props */
  fontSize: string,
  textDecoration: string,
  underline: bool,

  /* Built-In Props */
  children: node,
  href: string,
  variant: oneOf(Object.keys(VARIANT))
};
