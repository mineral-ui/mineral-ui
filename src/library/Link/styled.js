/* @flow */
import isPropValid from '@emotion/is-prop-valid';
import styled from '@emotion/styled';
import { linkTheme } from './themes';

export const Link = styled('a', {
  shouldForwardProp: (prop) => isPropValid(prop)
})(({ fontSize, textDecoration, underline, variant, theme: baseTheme }) => {
  let theme = linkTheme(baseTheme);

  if (variant) {
    // prettier-ignore
    theme = {
        ...theme,
        Link_borderColor_focus: baseTheme[`borderColor_${variant}_focus`],
        Link_color: baseTheme[`color_${variant}`],
        Link_color_hover: baseTheme[`color_${variant}_hover`],
        Link_color_focus: baseTheme[`color_${variant}_focus`],
        Link_color_active: baseTheme[`color_${variant}_active`],
      };
  }

  if (!textDecoration && underline) {
    textDecoration = 'underline';
  }

  return {
    color: theme.Link_color,
    textDecoration: 'none',

    '&:hover': {
      color: theme.Link_color_hover,
      textDecoration: textDecoration || 'underline'
    },
    '&:focus': {
      color: theme.Link_color_focus,
      outline: `1px solid ${theme.Link_borderColor_focus}`,
      outlineOffset: '2px'
    },
    // `:active` must be last, to follow LVHFA order:
    // https://developer.mozilla.org/en-US/docs/Web/CSS/:active
    '&:active': {
      color: theme.Link_color_active
    },

    /* TargetX Custom Styles */
    cursor: theme.Link_cursor,
    ...(fontSize && { fontSize }),
    ...(textDecoration && { textDecoration })
  };
});
