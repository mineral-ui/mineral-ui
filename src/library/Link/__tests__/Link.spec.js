/* @flow */
import React from 'react';
import { shallow } from 'enzyme';
import Link from '../Link';
import examples from '../../../website/app/demos/Link/examples';
import testDemoExamples from '../../../../utils/testDemoExamples';

function shallowLink(props) {
  return shallow(<Link {...props} />);
}

describe('Link', () => {
  it('renders', () => {
    const link = shallowLink({
      href: 'http://example.com',
      children: 'Children'
    });

    expect(link.exists()).toEqual(true);
  });

  testDemoExamples(examples, { exclude: ['react-router'] });
});
