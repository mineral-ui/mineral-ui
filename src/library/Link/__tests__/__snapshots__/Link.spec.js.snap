// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link demo examples basic 1`] = `
.css-0,
[data-css-0] {
  font-family: Open Sans;
}

.css-1,
[data-css-1] {
  font-size: 1.5em;
}

.css-2,
[data-css-2] {
  color: #2e6fd9;
  text-decoration: none;
}

.css-2:hover,
[data-css-2]:hover,
.css-2[data-simulate-hover],
[data-css-2][data-simulate-hover] {
  color: #4a89e8;
  text-decoration: underline;
}

.css-2:focus,
[data-css-2]:focus,
.css-2[data-simulate-focus],
[data-css-2][data-simulate-focus] {
  color: #2e6fd9;
  outline: 1px solid #2e6fd9;
  outline-offset: 2px;
}

.css-2:active,
[data-css-2]:active,
.css-2[data-simulate-active],
[data-css-2][data-simulate-active] {
  color: #1f5dc2;
}

.css-12,
[data-css-12] {
  font-size: 0.75em;
}

.css-13,
[data-css-13] {
  font-weight: 700;
}

.css-14,
[data-css-14] {
  font-family: serif;
}

<glamorous(div)>
  <div
    className="css-0"
  >
    <glamorous(p)>
      <p
        className="css-1"
      >
        Lorem ipsum dolor sit amet, 
        <Link
          href="http://example.com"
        >
          <Link
            href="http://example.com"
            target="_blank"
          >
            <Link
              href="http://example.com"
              target="_blank"
              variant="regular"
            >
              <a
                className="css-2"
                href="http://example.com"
                target="_blank"
              >
                consectetur
              </a>
            </Link>
          </Link>
        </Link>
         adipiscing elit,
      </p>
    </glamorous(p)>
    <glamorous(p)>
      <p
        className="css-12"
      >
        sed do eiusmod tempor 
        <Link
          href="http://example.com"
        >
          <Link
            href="http://example.com"
            target="_blank"
          >
            <Link
              href="http://example.com"
              target="_blank"
              variant="regular"
            >
              <a
                className="css-2"
                href="http://example.com"
                target="_blank"
              >
                incididunt
              </a>
            </Link>
          </Link>
        </Link>
         ut labore et dolore magna aliqua.
      </p>
    </glamorous(p)>
    <glamorous(p)>
      <p
        className="css-13"
      >
        Ut enim 
        <Link
          href="http://example.com"
        >
          <Link
            href="http://example.com"
            target="_blank"
          >
            <Link
              href="http://example.com"
              target="_blank"
              variant="regular"
            >
              <a
                className="css-2"
                href="http://example.com"
                target="_blank"
              >
                ad minim veniam
              </a>
            </Link>
          </Link>
        </Link>
         quis nostrud exercitation
      </p>
    </glamorous(p)>
    <glamorous(p)>
      <p
        className="css-14"
      >
        ullamco laboris 
        <Link
          href="http://example.com"
        >
          <Link
            href="http://example.com"
            target="_blank"
          >
            <Link
              href="http://example.com"
              target="_blank"
              variant="regular"
            >
              <a
                className="css-2"
                href="http://example.com"
                target="_blank"
              >
                nisi ut
              </a>
            </Link>
          </Link>
        </Link>
         aliquip ex ea commodo consequat.
      </p>
    </glamorous(p)>
  </div>
</glamorous(div)>
`;

exports[`Link demo examples button 1`] = `
.css-0,
[data-css-0] {
  box-sizing: border-box;
  color: #333840;
  font-family: Open Sans, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
  background-color: #ebeff5;
  border-color: #c8d1e0;
  border-radius: 0.1875em;
  border-style: solid;
  border-width: 1px;
  cursor: pointer;
  display: inline-block;
  font-weight: 600;
  height: 2.5em;
  min-width: 2.5em;
  padding: 0 0.5em;
  text-decoration: none;
  vertical-align: middle;
}

.css-0 *,
[data-css-0] *,
.css-0 *::before,
[data-css-0] *::before,
.css-0 *::after,
[data-css-0] *::after {
  box-sizing: inherit;
}

.css-0:focus,
[data-css-0]:focus,
.css-0[data-simulate-focus],
[data-css-0][data-simulate-focus] {
  background-color: #ebeff5;
  border-color: #fff;
  box-shadow: 0 0 0 1px #2e6fd9;
  color: #333840;
  text-decoration: none;
}

.css-0:hover,
[data-css-0]:hover,
.css-0[data-simulate-hover],
[data-css-0][data-simulate-hover] {
  background-color: #f5f7fa;
  color: #333840;
  text-decoration: none;
}

.css-0:active,
[data-css-0]:active,
.css-0[data-simulate-active],
[data-css-0][data-simulate-active] {
  background-color: #dde3ed;
  color: #333840;
}

.css-0::-moz-focus-inner,
[data-css-0]::-moz-focus-inner,
.css-0[data-simulate-mozfocusinner],
[data-css-0][data-simulate-mozfocusinner] {
  border: 0;
}

.css-0 [role="img"],
[data-css-0] [role="img"] {
  box-sizing: content-box;
  fill: #2e6fd9;
  display: block;
}

.css-0 [role="img"]:first-child,
[data-css-0] [role="img"]:first-child {
  margin-right: 0.5em;
}

.css-0 [role="img"]:last-child,
[data-css-0] [role="img"]:last-child {
  margin-left: 0.5em;
}

.css-0 [role="img"]:only-child,
[data-css-0] [role="img"]:only-child {
  margin: 0;
}

.css-14,
[data-css-14] {
  align-items: center;
  display: -webkit-inline-box;
  display: -moz-inline-box;
  display: -ms-inline-flexbox;
  display: -webkit-inline-flex;
  display: inline-flex;
  justify-content: center;
  max-height: 100%;
  width: 100%;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
}

.css-15,
[data-css-15] {
  display: block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  word-wrap: normal;
  font-size: 0.875em;
  line-height: 2.857142857142857em;
}

.css-15:first-child,
[data-css-15]:first-child,
.css-15[data-simulate-firstchild],
[data-css-15][data-simulate-firstchild] {
  padding-left: 0.5714285714285714em;
}

.css-15:last-child,
[data-css-15]:last-child,
.css-15[data-simulate-lastchild],
[data-css-15][data-simulate-lastchild] {
  padding-right: 0.5714285714285714em;
}

<Button
  element="a"
  href="#button"
  size="large"
  type="button"
  variant="regular"
>
  <glamorous(a)
    element="a"
    href="#button"
    size="large"
    text="Link"
    type="button"
    variant="regular"
  >
    <a
      className="css-0"
      href="#button"
    >
      <glamorous(span)>
        <span
          className="css-14"
        >
          <glamorous(span)
            size="large"
          >
            <span
              className="css-15"
            >
              Link
            </span>
          </glamorous(span)>
        </span>
      </glamorous(span)>
    </a>
  </glamorous(a)>
</Button>
`;

exports[`Link demo examples children 1`] = `
.css-0,
[data-css-0] {
  color: #2e6fd9;
  text-decoration: none;
}

.css-0:hover,
[data-css-0]:hover,
.css-0[data-simulate-hover],
[data-css-0][data-simulate-hover] {
  color: #4a89e8;
  text-decoration: underline;
}

.css-0:focus,
[data-css-0]:focus,
.css-0[data-simulate-focus],
[data-css-0][data-simulate-focus] {
  color: #2e6fd9;
  outline: 1px solid #2e6fd9;
  outline-offset: 2px;
}

.css-0:active,
[data-css-0]:active,
.css-0[data-simulate-active],
[data-css-0][data-simulate-active] {
  color: #1f5dc2;
}

<Link
  href="http://example.com"
>
  <Link
    href="http://example.com"
    target="_blank"
  >
    <Link
      href="http://example.com"
      target="_blank"
      variant="regular"
    >
      <a
        className="css-0"
        href="http://example.com"
        target="_blank"
      >
        <img
          alt="gradient placeholder"
          src="/images/125x125.png"
        />
      </a>
    </Link>
  </Link>
</Link>
`;

exports[`Link demo examples other 1`] = `
.css-0,
[data-css-0] {
  font-family: Open Sans;
}

.css-1,
[data-css-1] {
  color: #2e6fd9;
  text-decoration: none;
}

.css-1:hover,
[data-css-1]:hover,
.css-1[data-simulate-hover],
[data-css-1][data-simulate-hover] {
  color: #4a89e8;
  text-decoration: underline;
}

.css-1:focus,
[data-css-1]:focus,
.css-1[data-simulate-focus],
[data-css-1][data-simulate-focus] {
  color: #2e6fd9;
  outline: 1px solid #2e6fd9;
  outline-offset: 2px;
}

.css-1:active,
[data-css-1]:active,
.css-1[data-simulate-active],
[data-css-1][data-simulate-active] {
  color: #1f5dc2;
}

<glamorous(div)>
  <div
    className="css-0"
  >
    <BrowserRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "blank",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Link
          element={[Function]}
          to="/components/link"
        >
          <Link
            to="/components/link"
            variant="regular"
          >
            <Link
              className="css-1"
              replace={false}
              to="/components/link"
            >
              <a
                className="css-1"
                href="/components/link"
                onClick={[Function]}
              >
                ReactRouter Link
              </a>
            </Link>
          </Link>
        </Link>
      </Router>
    </BrowserRouter>
  </div>
</glamorous(div)>
`;

exports[`Link demo examples states 1`] = `
.css-0,
[data-css-0] {
  color: #2e6fd9;
  text-decoration: none;
}

.css-0:hover,
[data-css-0]:hover,
.css-0[data-simulate-hover],
[data-css-0][data-simulate-hover] {
  color: #4a89e8;
  text-decoration: underline;
}

.css-0:focus,
[data-css-0]:focus,
.css-0[data-simulate-focus],
[data-css-0][data-simulate-focus] {
  color: #2e6fd9;
  outline: 1px solid #2e6fd9;
  outline-offset: 2px;
}

.css-0:active,
[data-css-0]:active,
.css-0[data-simulate-active],
[data-css-0][data-simulate-active] {
  color: #1f5dc2;
}

.css-10,
[data-css-10] {
  font-family: Open Sans;
}

.css-10 > a,
[data-css-10] > a {
  margin-right: 0.5rem;
}

.css-11,
[data-css-11] {
  color: #db2929;
  text-decoration: none;
}

.css-11:hover,
[data-css-11]:hover,
.css-11[data-simulate-hover],
[data-css-11][data-simulate-hover] {
  color: #eb4d4d;
  text-decoration: underline;
}

.css-11:focus,
[data-css-11]:focus,
.css-11[data-simulate-focus],
[data-css-11][data-simulate-focus] {
  color: #db2929;
  outline: 1px solid #c71818;
  outline-offset: 2px;
}

.css-11:active,
[data-css-11]:active,
.css-11[data-simulate-active],
[data-css-11][data-simulate-active] {
  color: #c71818;
}

.css-21,
[data-css-21] {
  color: #06783f;
  text-decoration: none;
}

.css-21:hover,
[data-css-21]:hover,
.css-21[data-simulate-hover],
[data-css-21][data-simulate-hover] {
  color: #0a8f4c;
  text-decoration: underline;
}

.css-21:focus,
[data-css-21]:focus,
.css-21[data-simulate-focus],
[data-css-21][data-simulate-focus] {
  color: #06783f;
  outline: 1px solid #0a8f4c;
  outline-offset: 2px;
}

.css-21:active,
[data-css-21]:active,
.css-21[data-simulate-active],
[data-css-21][data-simulate-active] {
  color: #046132;
}

.css-31,
[data-css-31] {
  color: #cf4615;
  text-decoration: none;
}

.css-31:hover,
[data-css-31]:hover,
.css-31[data-simulate-hover],
[data-css-31][data-simulate-hover] {
  color: #e05b2b;
  text-decoration: underline;
}

.css-31:focus,
[data-css-31]:focus,
.css-31[data-simulate-focus],
[data-css-31][data-simulate-focus] {
  color: #cf4615;
  outline: 1px solid #cf4615;
  outline-offset: 2px;
}

.css-31:active,
[data-css-31]:active,
.css-31[data-simulate-active],
[data-css-31][data-simulate-active] {
  color: #bd3909;
}

<glamorous(glamorous(div))>
  <div
    className="css-10"
  >
    <Link
      href="http://example.com"
    >
      <Link
        href="http://example.com"
        target="_blank"
      >
        <Link
          href="http://example.com"
          target="_blank"
          variant="regular"
        >
          <a
            className="css-0"
            href="http://example.com"
            target="_blank"
          >
            Regular
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-hover=""
      href="http://example.com"
    >
      <Link
        data-simulate-hover=""
        href="http://example.com"
        target="_blank"
      >
        <Link
          data-simulate-hover=""
          href="http://example.com"
          target="_blank"
          variant="regular"
        >
          <a
            className="css-0"
            data-simulate-hover=""
            href="http://example.com"
            target="_blank"
          >
            Hover
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-focus=""
      href="http://example.com"
    >
      <Link
        data-simulate-focus=""
        href="http://example.com"
        target="_blank"
      >
        <Link
          data-simulate-focus=""
          href="http://example.com"
          target="_blank"
          variant="regular"
        >
          <a
            className="css-0"
            data-simulate-focus=""
            href="http://example.com"
            target="_blank"
          >
            Focus
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-focus=""
      data-simulate-hover=""
      href="http://example.com"
    >
      <Link
        data-simulate-focus=""
        data-simulate-hover=""
        href="http://example.com"
        target="_blank"
      >
        <Link
          data-simulate-focus=""
          data-simulate-hover=""
          href="http://example.com"
          target="_blank"
          variant="regular"
        >
          <a
            className="css-0"
            data-simulate-focus=""
            data-simulate-hover=""
            href="http://example.com"
            target="_blank"
          >
            Focus & Hover
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-active=""
      data-simulate-focus=""
      href="http://example.com"
    >
      <Link
        data-simulate-active=""
        data-simulate-focus=""
        href="http://example.com"
        target="_blank"
      >
        <Link
          data-simulate-active=""
          data-simulate-focus=""
          href="http://example.com"
          target="_blank"
          variant="regular"
        >
          <a
            className="css-0"
            data-simulate-active=""
            data-simulate-focus=""
            href="http://example.com"
            target="_blank"
          >
            Focus & Active
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-active=""
      href="http://example.com"
    >
      <Link
        data-simulate-active=""
        href="http://example.com"
        target="_blank"
      >
        <Link
          data-simulate-active=""
          href="http://example.com"
          target="_blank"
          variant="regular"
        >
          <a
            className="css-0"
            data-simulate-active=""
            href="http://example.com"
            target="_blank"
          >
            Active
          </a>
        </Link>
      </Link>
    </Link>
    <br />
    <br />
    <Link
      href="http://example.com"
      variant="danger"
    >
      <Link
        href="http://example.com"
        target="_blank"
        variant="danger"
      >
        <Link
          href="http://example.com"
          target="_blank"
          variant="danger"
        >
          <a
            className="css-11"
            href="http://example.com"
            target="_blank"
          >
            Danger
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-hover=""
      href="http://example.com"
      variant="danger"
    >
      <Link
        data-simulate-hover=""
        href="http://example.com"
        target="_blank"
        variant="danger"
      >
        <Link
          data-simulate-hover=""
          href="http://example.com"
          target="_blank"
          variant="danger"
        >
          <a
            className="css-11"
            data-simulate-hover=""
            href="http://example.com"
            target="_blank"
          >
            Hover
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-focus=""
      href="http://example.com"
      variant="danger"
    >
      <Link
        data-simulate-focus=""
        href="http://example.com"
        target="_blank"
        variant="danger"
      >
        <Link
          data-simulate-focus=""
          href="http://example.com"
          target="_blank"
          variant="danger"
        >
          <a
            className="css-11"
            data-simulate-focus=""
            href="http://example.com"
            target="_blank"
          >
            Focus
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-focus=""
      data-simulate-hover=""
      href="http://example.com"
      variant="danger"
    >
      <Link
        data-simulate-focus=""
        data-simulate-hover=""
        href="http://example.com"
        target="_blank"
        variant="danger"
      >
        <Link
          data-simulate-focus=""
          data-simulate-hover=""
          href="http://example.com"
          target="_blank"
          variant="danger"
        >
          <a
            className="css-11"
            data-simulate-focus=""
            data-simulate-hover=""
            href="http://example.com"
            target="_blank"
          >
            Focus & Hover
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-active=""
      data-simulate-focus=""
      href="http://example.com"
      variant="danger"
    >
      <Link
        data-simulate-active=""
        data-simulate-focus=""
        href="http://example.com"
        target="_blank"
        variant="danger"
      >
        <Link
          data-simulate-active=""
          data-simulate-focus=""
          href="http://example.com"
          target="_blank"
          variant="danger"
        >
          <a
            className="css-11"
            data-simulate-active=""
            data-simulate-focus=""
            href="http://example.com"
            target="_blank"
          >
            Focus & Active
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-active=""
      href="http://example.com"
      variant="danger"
    >
      <Link
        data-simulate-active=""
        href="http://example.com"
        target="_blank"
        variant="danger"
      >
        <Link
          data-simulate-active=""
          href="http://example.com"
          target="_blank"
          variant="danger"
        >
          <a
            className="css-11"
            data-simulate-active=""
            href="http://example.com"
            target="_blank"
          >
            Active
          </a>
        </Link>
      </Link>
    </Link>
    <br />
    <br />
    <Link
      href="http://example.com"
      variant="success"
    >
      <Link
        href="http://example.com"
        target="_blank"
        variant="success"
      >
        <Link
          href="http://example.com"
          target="_blank"
          variant="success"
        >
          <a
            className="css-21"
            href="http://example.com"
            target="_blank"
          >
            Success
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-hover=""
      href="http://example.com"
      variant="success"
    >
      <Link
        data-simulate-hover=""
        href="http://example.com"
        target="_blank"
        variant="success"
      >
        <Link
          data-simulate-hover=""
          href="http://example.com"
          target="_blank"
          variant="success"
        >
          <a
            className="css-21"
            data-simulate-hover=""
            href="http://example.com"
            target="_blank"
          >
            Hover
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-focus=""
      href="http://example.com"
      variant="success"
    >
      <Link
        data-simulate-focus=""
        href="http://example.com"
        target="_blank"
        variant="success"
      >
        <Link
          data-simulate-focus=""
          href="http://example.com"
          target="_blank"
          variant="success"
        >
          <a
            className="css-21"
            data-simulate-focus=""
            href="http://example.com"
            target="_blank"
          >
            Focus
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-focus=""
      data-simulate-hover=""
      href="http://example.com"
      variant="success"
    >
      <Link
        data-simulate-focus=""
        data-simulate-hover=""
        href="http://example.com"
        target="_blank"
        variant="success"
      >
        <Link
          data-simulate-focus=""
          data-simulate-hover=""
          href="http://example.com"
          target="_blank"
          variant="success"
        >
          <a
            className="css-21"
            data-simulate-focus=""
            data-simulate-hover=""
            href="http://example.com"
            target="_blank"
          >
            Focus & Hover
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-active=""
      data-simulate-focus=""
      href="http://example.com"
      variant="success"
    >
      <Link
        data-simulate-active=""
        data-simulate-focus=""
        href="http://example.com"
        target="_blank"
        variant="success"
      >
        <Link
          data-simulate-active=""
          data-simulate-focus=""
          href="http://example.com"
          target="_blank"
          variant="success"
        >
          <a
            className="css-21"
            data-simulate-active=""
            data-simulate-focus=""
            href="http://example.com"
            target="_blank"
          >
            Focus & Active
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-active=""
      href="http://example.com"
      variant="success"
    >
      <Link
        data-simulate-active=""
        href="http://example.com"
        target="_blank"
        variant="success"
      >
        <Link
          data-simulate-active=""
          href="http://example.com"
          target="_blank"
          variant="success"
        >
          <a
            className="css-21"
            data-simulate-active=""
            href="http://example.com"
            target="_blank"
          >
            Active
          </a>
        </Link>
      </Link>
    </Link>
    <br />
    <br />
    <Link
      href="http://example.com"
      variant="warning"
    >
      <Link
        href="http://example.com"
        target="_blank"
        variant="warning"
      >
        <Link
          href="http://example.com"
          target="_blank"
          variant="warning"
        >
          <a
            className="css-31"
            href="http://example.com"
            target="_blank"
          >
            Warning
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-hover=""
      href="http://example.com"
      variant="warning"
    >
      <Link
        data-simulate-hover=""
        href="http://example.com"
        target="_blank"
        variant="warning"
      >
        <Link
          data-simulate-hover=""
          href="http://example.com"
          target="_blank"
          variant="warning"
        >
          <a
            className="css-31"
            data-simulate-hover=""
            href="http://example.com"
            target="_blank"
          >
            Hover
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-focus=""
      href="http://example.com"
      variant="warning"
    >
      <Link
        data-simulate-focus=""
        href="http://example.com"
        target="_blank"
        variant="warning"
      >
        <Link
          data-simulate-focus=""
          href="http://example.com"
          target="_blank"
          variant="warning"
        >
          <a
            className="css-31"
            data-simulate-focus=""
            href="http://example.com"
            target="_blank"
          >
            Focus
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-focus=""
      data-simulate-hover=""
      href="http://example.com"
      variant="warning"
    >
      <Link
        data-simulate-focus=""
        data-simulate-hover=""
        href="http://example.com"
        target="_blank"
        variant="warning"
      >
        <Link
          data-simulate-focus=""
          data-simulate-hover=""
          href="http://example.com"
          target="_blank"
          variant="warning"
        >
          <a
            className="css-31"
            data-simulate-focus=""
            data-simulate-hover=""
            href="http://example.com"
            target="_blank"
          >
            Focus & Hover
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-active=""
      data-simulate-focus=""
      href="http://example.com"
      variant="warning"
    >
      <Link
        data-simulate-active=""
        data-simulate-focus=""
        href="http://example.com"
        target="_blank"
        variant="warning"
      >
        <Link
          data-simulate-active=""
          data-simulate-focus=""
          href="http://example.com"
          target="_blank"
          variant="warning"
        >
          <a
            className="css-31"
            data-simulate-active=""
            data-simulate-focus=""
            href="http://example.com"
            target="_blank"
          >
            Focus & Active
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      data-simulate-active=""
      href="http://example.com"
      variant="warning"
    >
      <Link
        data-simulate-active=""
        href="http://example.com"
        target="_blank"
        variant="warning"
      >
        <Link
          data-simulate-active=""
          href="http://example.com"
          target="_blank"
          variant="warning"
        >
          <a
            className="css-31"
            data-simulate-active=""
            href="http://example.com"
            target="_blank"
          >
            Active
          </a>
        </Link>
      </Link>
    </Link>
  </div>
</glamorous(glamorous(div))>
`;

exports[`Link demo examples variants 1`] = `
.css-0,
[data-css-0] {
  color: #2e6fd9;
  text-decoration: none;
}

.css-0:hover,
[data-css-0]:hover,
.css-0[data-simulate-hover],
[data-css-0][data-simulate-hover] {
  color: #4a89e8;
  text-decoration: underline;
}

.css-0:focus,
[data-css-0]:focus,
.css-0[data-simulate-focus],
[data-css-0][data-simulate-focus] {
  color: #2e6fd9;
  outline: 1px solid #2e6fd9;
  outline-offset: 2px;
}

.css-0:active,
[data-css-0]:active,
.css-0[data-simulate-active],
[data-css-0][data-simulate-active] {
  color: #1f5dc2;
}

.css-10,
[data-css-10] {
  font-family: Open Sans;
}

.css-10 > a,
[data-css-10] > a {
  margin-right: 0.5rem;
}

.css-11,
[data-css-11] {
  color: #db2929;
  text-decoration: none;
}

.css-11:hover,
[data-css-11]:hover,
.css-11[data-simulate-hover],
[data-css-11][data-simulate-hover] {
  color: #eb4d4d;
  text-decoration: underline;
}

.css-11:focus,
[data-css-11]:focus,
.css-11[data-simulate-focus],
[data-css-11][data-simulate-focus] {
  color: #db2929;
  outline: 1px solid #c71818;
  outline-offset: 2px;
}

.css-11:active,
[data-css-11]:active,
.css-11[data-simulate-active],
[data-css-11][data-simulate-active] {
  color: #c71818;
}

.css-21,
[data-css-21] {
  color: #06783f;
  text-decoration: none;
}

.css-21:hover,
[data-css-21]:hover,
.css-21[data-simulate-hover],
[data-css-21][data-simulate-hover] {
  color: #0a8f4c;
  text-decoration: underline;
}

.css-21:focus,
[data-css-21]:focus,
.css-21[data-simulate-focus],
[data-css-21][data-simulate-focus] {
  color: #06783f;
  outline: 1px solid #0a8f4c;
  outline-offset: 2px;
}

.css-21:active,
[data-css-21]:active,
.css-21[data-simulate-active],
[data-css-21][data-simulate-active] {
  color: #046132;
}

.css-31,
[data-css-31] {
  color: #cf4615;
  text-decoration: none;
}

.css-31:hover,
[data-css-31]:hover,
.css-31[data-simulate-hover],
[data-css-31][data-simulate-hover] {
  color: #e05b2b;
  text-decoration: underline;
}

.css-31:focus,
[data-css-31]:focus,
.css-31[data-simulate-focus],
[data-css-31][data-simulate-focus] {
  color: #cf4615;
  outline: 1px solid #cf4615;
  outline-offset: 2px;
}

.css-31:active,
[data-css-31]:active,
.css-31[data-simulate-active],
[data-css-31][data-simulate-active] {
  color: #bd3909;
}

<glamorous(glamorous(div))>
  <div
    className="css-10"
  >
    <Link
      href="http://example.com"
    >
      <Link
        href="http://example.com"
        target="_blank"
      >
        <Link
          href="http://example.com"
          target="_blank"
          variant="regular"
        >
          <a
            className="css-0"
            href="http://example.com"
            target="_blank"
          >
            Regular
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      href="http://example.com"
      variant="danger"
    >
      <Link
        href="http://example.com"
        target="_blank"
        variant="danger"
      >
        <Link
          href="http://example.com"
          target="_blank"
          variant="danger"
        >
          <a
            className="css-11"
            href="http://example.com"
            target="_blank"
          >
            Danger
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      href="http://example.com"
      variant="success"
    >
      <Link
        href="http://example.com"
        target="_blank"
        variant="success"
      >
        <Link
          href="http://example.com"
          target="_blank"
          variant="success"
        >
          <a
            className="css-21"
            href="http://example.com"
            target="_blank"
          >
            Success
          </a>
        </Link>
      </Link>
    </Link>
    <Link
      href="http://example.com"
      variant="warning"
    >
      <Link
        href="http://example.com"
        target="_blank"
        variant="warning"
      >
        <Link
          href="http://example.com"
          target="_blank"
          variant="warning"
        >
          <a
            className="css-31"
            href="http://example.com"
            target="_blank"
          >
            Warning
          </a>
        </Link>
      </Link>
    </Link>
  </div>
</glamorous(glamorous(div))>
`;
